src/backend/distributed/commands/extension.c

```diff
@@ -115,3 +110,516 @@ ExtractNewExtensionVersion(Node *parsetree)
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return;
+	}
+
+	extensionAddress = GetObjectAddressFromParseTree((Node *) createExtensionStmt, false);
+
+	EnsureDependenciesExistsOnAllNodes(extensionAddress);
+
+	MarkObjectDistributed(extensionAddress);
+}
+
+
+/*
+ * PlanDropExtensionStmt is called to drop extension(s) in coordinator and
+ * in worker nodes if distributed before.
+ * We first ensure that we keep only the distributed ones before propagating
+ * the statement to worker nodes.
+ * If no extensions in the drop list are distributed, then no calls will
+ * be made to the workers.
+ */
+List *
+PlanDropExtensionStmt(DropStmt *dropStmt, const char *queryString)
+{
+	List *oldObjects = dropStmt->objects;
+
+	List *distributedExtensions = NIL;
+	List *distributedExtensionAddresses = NIL;
+
+	List *commands = NIL;
+	const char *deparsedStmt = NULL;
+
+	ListCell *addressCell = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) dropStmt))
+	{
+		return NIL;
+	}
+
+	/* Get distributed extensions to be dropped in worker nodes as well */
+	distributedExtensions = FilterDistributedExtensions(oldObjects);
+
+	if (list_length(distributedExtensions) <= 0)
+	{
+		/* no distributed extensions to drop */
+		return NIL;
+	}
+
+	/* Extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	distributedExtensionAddresses = ExtensionNameListToObjectAddresses(
+		distributedExtensions);
+
+	/* Unmark each distributed extension */
+	foreach(addressCell, distributedExtensionAddresses)
+	{
+		ObjectAddress *address = (ObjectAddress *) lfirst(addressCell);
+		UnmarkObjectDistributed(address);
+	}
+
+	/*
+	 * Temporary swap the lists of objects to delete with the distributed
+	 * objects and deparse to an sql statement for the workers.
+	 * Then switch back to oldObjects to drop all specified extensions in
+	 * coordinator after PlanDropExtensionStmt completes its execution.
+	 */
+	dropStmt->objects = distributedExtensions;
+	deparsedStmt = DeparseTreeNode((Node *) dropStmt);
+
+	dropStmt->objects = oldObjects;
+
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) deparsedStmt,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * Given the "objects" list of a DropStmt, return the distributed ones in
+ * a Value(String) list.
+ */
+static List *
+FilterDistributedExtensions(List *objects)
+{
+	List *result = NIL;
+
+	bool missingOk = true;
+	ListCell *objectCell = NULL;
+
+	foreach(objectCell, objects)
+	{
+		char *extensionName = strVal(lfirst(objectCell));
+
+		ObjectAddress *address = palloc0(sizeof(ObjectAddress));
+
+		Oid extensionoid = get_extension_oid(extensionName, missingOk);
+
+		if (!OidIsValid(extensionoid))
+		{
+			continue;
+		}
+
+		ObjectAddressSet(*address, ExtensionRelationId, extensionoid);
+
+		if (!IsObjectDistributed(address))
+		{
+			continue;
+		}
+
+		result = lappend(result, makeString(extensionName));
```

Comment:
why do we use `makeString` here? Can't we simply add `extensionName` to list.

Because we do it here, we have to use `strVal` later on to get the `char *`, let's simplify by removing both

Reply:
Actually, I aimed to implement `ExtensionNameListToObjectAddresses` `(ExtensionNameListToObjectAddressList)` function to be a generic one so it takes an `object` list of a `DropStmt`. This is why I am returning a list of `makeString(char*)` (which returns `Value`) in `FilterDistributedExtensions`.

Does this approach seem to be a valid one ?